source/dynamical_systems/src/DynamicalSystem.cpp

```diff
@@ -19,6 +20,17 @@ template<>
 DynamicalSystem<CartesianState>::DynamicalSystem(const std::string& base_frame) :
     base_frame_(CartesianState::Identity(base_frame, base_frame)) {}
 
+template<class S>
+bool DynamicalSystem<S>::is_compatible(const S&) {
+  throw exceptions::NotImplementedException("is_compatible(state) not implemented for this type of state");
+}
+
+template<>
+bool DynamicalSystem<state_representation::CartesianState>::is_compatible(const state_representation::CartesianState& state) {
+  return !(state.get_reference_frame() != this->get_base_frame().get_name()
```

Comment:
This means that the state reference frame has to be equal to either the name or the reference frame of the DS base frame for it to be compatible right?